general/src/types/annotated.rs

```diff
@@ -54,55 +56,23 @@ impl MetaTree {
 /// Meta for children.
 pub type MetaMap = Map<String, MetaTree>;
 
-/// Used to indicate how to handle an annotated value in a callback.
-#[must_use = "This `ValueAction` must be handled by `Annotated::apply`"]
-#[derive(Copy, Clone, Debug, PartialEq, Eq)]
-pub enum ValueAction {
-    /// Keeps the value as is.
-    Keep,
+pub type ValueAction = Result<(), DiscardValue>;
```

Comment:
As per offline conversation

Rename to `ProcessingResult`. With the other changes from below it then becomes `ProcessingResult = Result<(), ProcessingAction>`;